### DISCUSSION TITLE
Data binding for godot gui?

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [godotengine/godot-proposals#9251] [GUI]
### Describe the project you are working on

A very gui heavy desktop editor for a mario game

### Describe the problem or limitation you are having in your project

Dependency explosion and code ending up in a knotted mess

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allow use to binding properties to other properties of the same type similar to modern gui frameworks

The idea is that we could decouple our model from our view

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Idk I'm not a gui developer, I would wait for other peoples input on this or look how other frameworks do it.
I just know that godot's gui system is too tangled for me to use right now.

Maybe you guys can look at avalonia ui for C# on how they do it.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Idk, I definitely don't know how, it would be nice if it was supported by the engine and encuraged

### Is there a reason why this should be core and not an add-on in the asset library?

It would improve code quality and decouple models from views.

**Comment by huwpascoe** [comment]
Can you give some examples of the workflow you'd like to be for binding? How would you make something in avalonia?

Also have a look at this excellent [tutorial](https://youtu.be/4vAkTHeoORk?t=1216&feature=shared) Incase you haven't already seen it (the link starts at the point for UI but I recommend watching the whole thing if you have time).

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.
